Guide/jobs.markdown

```diff
@@ -0,0 +1,89 @@
+# Jobs
+
+```toc
+
+```
+
+## Introduction
+
+IHP has built-in functionality for creating and running background jobs. Jobs are perfect for any tasks that can be split up into small units and run in parallel, such periodically cleaning the database, sending emails, and scraping data.
+
+### Creating a job
+
+In the codegen tool in the IHP IDE, use the "Background Job" option to generate the code for a new job. To illustrate the features of jobs, let's
+create a job to send an email to our application's customers.
+
+### Implementing the job
+
+The job file is created at `<Application>/Job/<Name>.hs`, and looks like:
+
+```haskell
+module Web.Job.EmailCustomers where
+import Web.Controller.Prelude
+
+instance Job EmailCustomersJob where
+    perform EmailCustomersJob { .. } = do
+        putStrLn "Hello World!"
+```
+
+The type `EmailCustomersJob` is a generated type based on the table the background job codegen added to `Schema.sql`. If you want to include additional data in the job, you can add rows to the table in the IHP IDE just like any custom table.
+
+A pseudocode implementation of a job that sends a marketing email to all of the customers:
+
+```haskell
+module Web.Job.EmailCustomers where
+import Web.Controller.Prelude
+
+instance Job EmailCustomersJob where
+    perform EmailCustomersJob { .. } = do
+      customers <- query @Customer |> fetch
+      mapM_ (\customer -> sendMail (MarketingMail customer)) customers
```

Comment:
I'd use `forEach` here which is like mapM_ but arguments reversed. This feels more like the IHP approach :)

Reply:
Never thought of using `forEach` outside of views! Thanks